### PR TITLE
rad-api-tool b0

### PR DESCRIPTION
**Pull request by advolkov**

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -0,0 +1,24 @@
+{
+    "name": "opensoft/rad-api-tool",
+    "description": "rad-api-tool",
```
Comment (eduard-sukharev) [inline composer.json]: For proper packaging description should be more.. descriptive :) Use the one from repository description:
`Tool aimed to aid in rapid api development.`

File: composer.json [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+            "email": "eduard.sukharev@opensoftdev.ru"
+        }
+    ],
+    "require": {
+        "php": ">=5.3.2",
+        "symfony/console": "2.3.*"
+    },
+    "autoload": {
+        "psr-0": {
+            "": "src/"
```
Comment (eduard-sukharev) [inline composer.json]: We should define our namespace here, e.g.:
`"Opensoft\RadApiTool" : "src/"`

File: src/Opensoft/RadApiTool/Command/RoutesGeneratorCommand.php [hunk 3]
```diff
@@ -0,0 +1,57 @@
+<?php
+/*
+ * This file is part of ONP Behat.
```
Comment (eduard-sukharev) [inline src/Opensoft/RadApiTool/Command/RoutesGeneratorCommand.php]: Behat? Oh really? ;-)

Comment (advolkov) [inline src/Opensoft/RadApiTool/Command/RoutesGeneratorCommand.php]: lol phpstorm autofill

Comment (eduard-sukharev) [inline src/Opensoft/RadApiTool/Command/RoutesGeneratorCommand.php]: Maybe we should move copyright to separate file and just say something like `see copyright in LICENCE.txt`?

File: src/Opensoft/RadApiTool/Command/RoutesGeneratorCommand.php [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+ *
+ * Copyright (c) 2013 Opensoft (http://opensoftdev.com)
+ *
+ * The unauthorized use of this code outside the boundaries of
+ * Opensoft is prohibited.
+ */
+
+namespace Opensoft\RadApiTool\Command;
+
+use Opensoft\RadApiTool\RoutesGenerator;
```
Comment (eduard-sukharev) [inline src/Opensoft/RadApiTool/Command/RoutesGeneratorCommand.php]: External dependencies first, then our own classes. Move this `use` line down to the bottom of use section

File: src/Opensoft/RadApiTool/Command/RoutesGeneratorCommand.php (hunk 3 above)
Comment (advolkov) [inline src/Opensoft/RadApiTool/Command/RoutesGeneratorCommand.php]: Yes, please commit this file

File: composer.json [hunk 5]
```diff
@@ -0,0 +1,24 @@
+{
+    "name": "opensoft/rad-api-tool",
+    "description": "rad-api-tool",
+    "license": "proprietary",
```
Comment (eduard-sukharev) [inline composer.json]: probably we should use some less strict license. LGPL? BSD?

File: src/Opensoft/RadApiTool/RoutesGenerator.php [hunk 6]
```diff
@@ -0,0 +1,81 @@
...
+ *
+ * The unauthorized use of this code outside the boundaries of
+ * Opensoft is prohibited.
+ */
+
+namespace Opensoft\RadApiTool;
+
+use Opensoft\RadApiTool\Utils;
+use Opensoft\RadApiTool\Command\RoutesGeneratorCommand;
+
```
Comment (eduard-sukharev) [inline src/Opensoft/RadApiTool/RoutesGenerator.php:16]: CS, extra empty line

File: src/Opensoft/RadApiTool/Command/RoutesGeneratorCommand.php [hunk 7]
```diff
@@ -0,0 +1,57 @@
...
+use Symfony\Component\Console\Command\Command;
+use Symfony\Component\Console\Input\InputArgument;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Input\InputOption;
+use Symfony\Component\Console\Output\OutputInterface;
+
+class RoutesGeneratorCommand extends Command
```
Comment (eduard-sukharev) [inline src/Opensoft/RadApiTool/Command/RoutesGeneratorCommand.php:24]: Add class phpdoc

File: src/Opensoft/RadApiTool/Command/RoutesGeneratorCommand.php [hunk 8]
```diff
@@ -0,0 +1,57 @@
...
+                "Entity name in provided bundle"
+            )
+            ->addArgument(
+                'base_url',
+                InputArgument::REQUIRED,
+                "Your company name"
+            );
+    }
+
+    protected function execute(InputInterface $input, OutputInterface $output)
```
Comment (eduard-sukharev) [inline src/Opensoft/RadApiTool/Command/RoutesGeneratorCommand.php:56]: phpdoc

File: src/Opensoft/RadApiTool/Utils.php [hunk 9]
```diff
@@ -0,0 +1,46 @@
...
+ */
+class Utils
+{
+    /**
+     * @param $name
+     * @return string
+     */
+    public static function toUnderscore($name)
+    {
+        if (stristr($name, 'bundle')) {
```
Comment (eduard-sukharev) [inline src/Opensoft/RadApiTool/Utils.php]: We should move this check and related string transformations out of this method, probably into Command itself.

File: src/Opensoft/RadApiTool/Utils.php [hunk 10]
```diff
@@ -0,0 +1,46 @@
...
+/**
+ * @author Anton Volkov <anton.volkov@opensoftdev.ru>
+ */
+class Utils
+{
+    /**
+     * @param $name
+     * @return string
+     */
+    public static function toUnderscore($name)
```
Comment (eduard-sukharev) [inline src/Opensoft/RadApiTool/Utils.php]: i guess proper name for `$name` would be something more generic, like `$string`. don't forget to update phpdoc accordingly

File: src/Opensoft/RadApiTool/Utils.php [hunk 11]
```diff
@@ -0,0 +1,46 @@
...
+        }
+
+        return implode('_', $result[0]);
+    }
+
+    /**
+     * @param $name
+     * @return mixed
+     */
+    public static function toCamelCase($name)
```
Comment (eduard-sukharev) [inline src/Opensoft/RadApiTool/Utils.php]: Same as above

File: src/Opensoft/RadApiTool/rad-api-tool.php [hunk 12]
```diff
@@ -0,0 +1,17 @@
...
+date_default_timezone_set('UTC');
+
+set_time_limit(0);
+
+(@include_once __DIR__ . '/../../../vendor/autoload.php');
+
+use Opensoft\RadApiTool\Command\RoutesGeneratorCommand;
+use Symfony\Component\Console\Application;
+use Opensoft\RadApiTool\RoutesGenerator;
+
```
Comment (eduard-sukharev) [inline src/Opensoft/RadApiTool/rad-api-tool.php:13]: CS, extra empty line

Comment (advolkov) [inline src/Opensoft/RadApiTool/rad-api-tool.php:13]: removed

File: src/Opensoft/RadApiTool/RoutesGenerator.php (hunk 6 above)
Comment (advolkov) [inline src/Opensoft/RadApiTool/RoutesGenerator.php:16]: removed

File: composer.json (hunk 1 above)
Comment (advolkov) [inline composer.json]: added

File: composer.json (hunk 2 above)
Comment (advolkov) [inline composer.json]: added

File: src/Opensoft/RadApiTool/Command/RoutesGeneratorCommand.php (hunk 4 above)
Comment (advolkov) [inline src/Opensoft/RadApiTool/Command/RoutesGeneratorCommand.php]: fixed

File: composer.json (hunk 5 above)
Comment (advolkov) [inline composer.json]: I don't know, we need to make decision

File: src/Opensoft/RadApiTool/Command/RoutesGeneratorCommand.php (hunk 7 above)
Comment (advolkov) [inline src/Opensoft/RadApiTool/Command/RoutesGeneratorCommand.php:24]: added

File: src/Opensoft/RadApiTool/Command/RoutesGeneratorCommand.php (hunk 8 above)
Comment (advolkov) [inline src/Opensoft/RadApiTool/Command/RoutesGeneratorCommand.php:56]: added

File: src/Opensoft/RadApiTool/Utils.php (hunk 10 above)
Comment (advolkov) [inline src/Opensoft/RadApiTool/Utils.php]: fixed

File: src/Opensoft/RadApiTool/Utils.php (hunk 11 above)
Comment (advolkov) [inline src/Opensoft/RadApiTool/Utils.php]: fixed

File: src/Opensoft/RadApiTool/Utils.php (hunk 9 above)
Comment (advolkov) [inline src/Opensoft/RadApiTool/Utils.php]: fixed

File: src/Opensoft/RadApiTool/rad-api-tool.php [hunk 13]
```diff
@@ -0,0 +1,16 @@
...
+<?php
+
+date_default_timezone_set('UTC');
+
+set_time_limit(0);
+
+(@include_once __DIR__ . '/../../../vendor/autoload.php');
+
+use Opensoft\RadApiTool\Command\RoutesGeneratorCommand;
+use Symfony\Component\Console\Application;
```
Comment (eduard-sukharev) [inline src/Opensoft/RadApiTool/rad-api-tool.php:10]: CS, reorder pls

**Pull request merged**